internal/integration/base/k8s.go

```diff
@@ -0,0 +1,65 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+// +build integration_k8s
+
+package base
+
+import (
+	"archive/tar"
+	"bytes"
+	"compress/gzip"
+	"context"
+	"io"
+
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+	clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
+)
+
+// K8sSuite is a base suite for K8s tests
+type K8sSuite struct {
+	APISuite
+
+	Clientset *kubernetes.Clientset
+}
+
+// SetupSuite initializes Kubernetes client
+func (k8sSuite *K8sSuite) SetupSuite() {
+	k8sSuite.APISuite.SetupSuite()
+
+	r, errCh, err := k8sSuite.Client.Kubeconfig(context.Background())
```

Comment:
I feel our `Kubeconfig` method is not really easy to use, as it requires reading .tar.gz and extracting files back. I thought of adding another method which return simple `[]byte` (based on code below), thoughts?

Reply:
Oof. Yeah this is pretty nasty for a simple API. Agree we should revert to just reading the file.

Reply:
I'll update the client code to wrap back .tar.gz stream to a single file so that we could more easily get what we need

Reply:
Added `client.Kubeconfig` as easy wrapper around `client.KubeconfigRaw` which provides streaming